### DISCUSSION TITLE
Dynamic features with unknown future values for prediction and cross-validation

### DISCUSSION THREAD

**Discussion by RashidBakirov** [Nixtla/mlforecast#144] [Q&A] [answered]
How to deal with features which are not static, but we don't know their values into the future? Would we just specify the static features in fit method and skip the dynamic_dfs argument in predict? Also, how do such dynamic features work for cross-validation? From the code it seems that all the features except target_col are treated as dynamic_dfs for cross-validation? 

EDIT: It looks like a feature must be specified either as static or dynamic, as specifying some static features in fit and no dynamic features in predict results in error. For cross-validation, all the features which are not explicitly static are treated as dynamic. Does this mean that there is currently dynamic features with unknown future values are not supported?

EDIT2: I guess one workaround is to shift these dynamic features with unknown future values forward by the forecasting window length, essentially creating artificial future values, and then including in predict with dynamic_dfs attribute.

**Comment by jmoralez** [comment, marked as answer]
Hi. It seems that you've answered most of your questions, except for the cross validation one. What we do there is that since we know the "future" values of dynamic features you don't have to provide the dynamic_dfs argument, we just use the ones from each validation set.

Please let us know if you have further doubts.
